### PR TITLE
Replace gorm with pop for db access

### PR DESCRIPTION
**Pull request by brycekahle**
**- Summary**

`gorm` uses a naive update method which sets all columns. This will cause updates to clobber each other if run in close succession. We are switching to a simpler ORM `pop` as a first step, and then replacing all updates with explicit column mentions.

Change model IDs to UUIDs.
Remove `deleted_at` from instances model.

**- Test plan**

All existing tests pass.

**- Description for the changelog**

Replace gorm with pop.

### CODE REVIEW

File: storage/storage.go [hunk 1]
```diff
@@ -1,31 +1,36 @@
 package storage
 
-import "github.com/netlify/gotrue/models"
+import (
+	"github.com/netlify/gotrue/models"
+	uuid "github.com/satori/go.uuid"
+)
 
 // Connection is the interface a storage provider must implement.
 type Connection interface {
```
Comment (rybit) [inline storage/storage.go]: we could cut this whole interface now that we're not supporting other than mysql interfaces right?

File: api/api_test.go [hunk 2]
```diff
@@ -44,34 +50,65 @@ func setupAPIForTestForInstance() (*API, *conf.Configuration, string, error) {
...
 
-func setupAPIForTestWithCallback(cb func(*conf.GlobalConfiguration, *conf.Configuration, storage.Connection) (string, error)) (*API, *conf.Configuration, error) {
-	globalConfig, conn, err := test.SetupDBConnection()
+func setupAPIForTestWithCallback(cb func(*conf.GlobalConfiguration, *conf.Configuration, storage.Connection) (uuid.UUID, error)) (*API, *conf.Configuration, error) {
+	globalConfig, err := conf.LoadGlobal(apiTestConfig)
 	if err != nil {
 		return nil, nil, err
 	}
 
+	// TODO disable serial tests and re-enable automigrate and uncomment
```
[the remark this answers is not in the document]
Comment (brycekahle) [reply api/api_test.go]: I did, but he hasn't released a new version yet. I could use a specific revision if we really want.

File: storage/storage.go (hunk 1 above)
Comment (brycekahle) [reply storage/storage.go]: We could, but it does provide a nice decoupling point.

File: conf/configuration.go [hunk 3]
```diff
@@ -21,11 +23,10 @@ type OAuthProviderConfiguration struct {
...
 type DBConfiguration struct {
-	Dialect     string `json:"dialect"`
-	Driver      string `json:"driver" required:"true"`
-	URL         string `json:"url" envconfig:"DATABASE_URL" required:"true"`
-	Namespace   string `json:"namespace"`
-	Automigrate bool   `json:"automigrate"`
+	Driver    string `json:"driver" required:"true"`
+	URL       string `json:"url" envconfig:"DATABASE_URL" required:"true"`
+	Namespace string `json:"namespace"`
+	// Automigrate bool   `json:"automigrate"`
```
[the remark this answers is not in the document]
Comment (brycekahle) [reply conf/configuration.go]: I think we want to support it, but some thought and work on how is necessary.

File: api/api_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (rybit) [reply api/api_test.go]: perfect - I don't think we need it for the moment. I think if we cut all the relevant code to this we could file an issue against us and then think it out a bit more

File: storage/storage.go (hunk 1 above)
Comment (rybit) [reply storage/storage.go]: sweet - if you think it helps, I'm fine with leaving it.

**Pull request merged**